src/Common/MSBuildCachePluginBase.cs

```diff
@@ -338,7 +338,8 @@ private async Task<CacheResult> GetCacheResultInnerAsync(BuildRequestData buildR
 
         nodeContext.SetStartTime();
 
-        if (!nodeContext.TargetNames.SetEquals(buildRequest.TargetNames))
+        bool isBuildAndTestRequest = IsBuildAndTestRequest(buildRequest);
+        if (!nodeContext.TargetNames.SetEquals(buildRequest.TargetNames) && !isBuildAndTestRequest)
```

Comment:
If the target names don't align, this seems like a correctness issue

Reply:
The targets don't align, MSBuildCache seems to ignore the Test target

Reply:
https://github.com/dotnet/msbuild/pull/9569